x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/components/generated_config_fields.tsx

```diff
@@ -96,6 +99,33 @@ export const GeneratedConfigFields: React.FC<GeneratedConfigFieldsProps> = ({
     if (generateApiKey) generateApiKey();
     setIsModalVisible(false);
   };
+  const searchIndexDetailsUrl = share?.url.locators
+    .get('SEARCH_INDEX_DETAILS_LOCATOR_ID')
+    ?.useUrl({ indexName: connector?.index_name });
```

Comment:
why is this usage of the locator so different from the others? if we can make a sync call that returns the value or `undefined` we should likely prefer this pattern.

Reply:
This usage didn't work with my previous approach but I agree useUrl is better. 
Updated in 6c03f45068133d97067aef5429bb53e1c4c97990 to use `useUrl` in both connector & search application